spark/interpreter/src/main/java/org/apache/zeppelin/spark/IPySparkInterpreter.java

```diff
@@ -51,9 +51,10 @@ public void open() throws InterpreterException {
         PySparkInterpreter.getPythonExec(getProperties()));
     sparkInterpreter = getSparkInterpreter();
     SparkConf conf = sparkInterpreter.getSparkContext().getConf();
-    // only set PYTHONPATH in local or yarn-client mode.
+    // only set PYTHONPATH in embedded, local or yarn-client mode.
     // yarn-cluster will setup PYTHONPATH automatically.
-    if (!conf.get("spark.submit.deployMode").equals("cluster")) {
+    if (!conf.contains("spark.submit.deployMode") ||
+        !conf.get("spark.submit.deployMode").equals("cluster")) {
```

Comment:
I think it's better to whitelist instead of a blacklist. also we should check the `master` - other places people are using with mesos, kubernetes etc.